lib/nitcorn/http_request_buffer.nit

```diff
@@ -12,29 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Http request parsing for bufferized inputs.
-module http_request_parser
+# Http request parsing for buffered inputs.
+module http_request_buffer
 
 intrude import libevent
 
-redef class Connection
+# Connection rebuilding HTTP requests
+#
+# Subclass should refine `read_full_request` and avoid `read_callback`.
+class HTTPConnection
+	super Connection
 
 	private var in_request = false
 	private var in_header = false
 	private var in_body = false
-	private var current_header: FlatBuffer
-	private var current_body: FlatBuffer
+	private var current_header = new Array[String]
+	private var current_body = new Array[String]
```

Comment:
Instead of a `Array[String]` what about a `Template` that is a higer version of this and that accepts Writable (thus Bytes)